### PR TITLE
Feat/add auth

### PR DESCRIPTION
**Pull request by Maianki**
## Added auth functionality and toast

- [ ] Auth functionality
- [ ] Toast

### Thanks for reviewing my PR

### CODE REVIEW

File: src/components/RestrictAuth/RestrictAuth.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export function RestrictAuth() {
+  const location = useLocation();
+  const { userToken } = useSelector((state) => state.auth);
+
+  return userToken ? (
+    <Navigate
+      to={`${
+        location?.state?.from?.pathname
+          ? location?.state?.from?.pathname
+          : "/home"
```
Comment (anik31) [inline src/components/RestrictAuth/RestrictAuth.jsx:14]: suggestion - you can use short circuiting or nullish coalescing here

File: src/features/Auth/Login.jsx [hunk 2]
```diff
@@ -12,9 +13,54 @@ import {
...
+
+  const changeHandler = (e) => {
+    setUserDetails({ ...userDetails, [e.target.name]: e.target.value });
+  };
+
+  const loginHandler = (e) => {
+    e.preventDefault();
+    if (userDetails.email === "" || userDetails.password === "") {
+      toast.error("Please enter both email and password", {
+        position: "top-right",
```
Comment (sainath7878) [inline src/features/Auth/Login.jsx]: suggestion: You can define the props of react toastify in ```ToastContainer``` itself

File: src/features/Auth/authSlice.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  authError: "",
+  authStatus: "idle",
+};
+
+export const loginUser = createAsyncThunk(
+  "auth/loginUser",
+  async ({ email, password }, { rejectWithValue }) => {
+    try {
+      const response = await loginService(email, password);
+      console.log(response);
```
Comment (sainath7878) [inline src/features/Auth/authSlice.js]: remove console.log

File: src/utils/constants/APIEndpoints.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export const SIGNUP_API = "/api/auth/signup";
```
Comment (sainath7878) [inline src/utils/constants/APIEndpoints.js:1]: Appreciation: Good that you're defining constants for string values!

**Review by sainath7878 — APPROVED** [review]
Everything looks good apart from the suggested change. Awesome UI.

File: src/features/Auth/authSlice.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      state.authStatus = "loading";
+    },
+    [loginUser.fulfilled]: (state, action) => {
+      state.authStatus = "success";
+      console.log("success here");
+      state.userToken = action.payload.encodedToken;
+      state.user = action.payload.foundUser;
+    },
+    [loginUser.rejected]: (state, action) => {
+      console.log("here");
```
Comment (anik31) [inline src/features/Auth/authSlice.js]: suggestion - remove console.log

**Review by anik31 — APPROVED** [review]
Nice work, the auth functionality is working fine. But I think you haven't added logout functionality.

File: src/features/Auth/authSlice.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  async (userDetails, { rejectWithValue }) => {
+    try {
+      const response = await signupService(userDetails);
+      if (response.status === 201) {
+        console.log(response.data);
+      }
+      return response.data;
+    } catch (error) {
+      const { response } = error;
+      console.log(error.response);
```
Comment (sameep-s) [inline src/features/Auth/authSlice.js:48]: you can remove console.log as you are using taost, or you can use console.error.

File: src/features/Auth/authSlice.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+  }
+);
+
+export const signupUser = createAsyncThunk(
+  "auth/signupUser",
+  async (userDetails, { rejectWithValue }) => {
+    try {
+      const response = await signupService(userDetails);
+      if (response.status === 201) {
+        console.log(response.data);
```
Comment (sameep-s) [inline src/features/Auth/authSlice.js:43]: you can remove this too.

**Review by sameep-s — APPROVED** [review]
- Like the folder structure.
- Good naming convention.
- Code is clean and well written.
- There are alot of unwanted console.log's you can remove them, other-wise code is well written.

File: src/components/RestrictAuth/RestrictAuth.jsx (hunk 1 above)
Comment (Maianki) [reply src/components/RestrictAuth/RestrictAuth.jsx:14]: Thanks will do that.

File: src/features/Auth/Login.jsx (hunk 2 above)
Comment (Maianki) [reply src/features/Auth/Login.jsx]: Thanks for telling I did not think of it. Will do that.

File: src/features/Auth/authSlice.js (hunk 7 above)
Comment (Maianki) [reply src/features/Auth/authSlice.js:43]: Thanks for pointing out will do that.

**Comment by Maianki** [comment]
Thanks @sainath7878 @anik31 @sameep-s  fro PR review

**Pull request merged**